src/tools/dev/scripts/bv_support/bv_python.sh

```diff
@@ -80,7 +80,11 @@ function python_set_vars_helper
     #remove any extra includes
     PYTHON_INCLUDE_PATH="${PYTHON_INCLUDE_PATH%%-I*}"
     PYTHON_INCLUDE_DIR="$PYTHON_INCLUDE_PATH"
-    PYTHON_LIBRARY=`"$PYTHON_CONFIG_COMMAND" --libs`
+    if [[ "$DO_PYTHON2" == "yes" ]] ; then
+        PYTHON_LIBRARY=`"$PYTHON_CONFIG_COMMAND" --libs`
+    else
+        PYTHON_LIBRARY=`"$PYTHON_CONFIG_COMMAND" --libs --embed`
```

Comment:
--embed doesn't work on all systems (just tried on our LC system).

Reply:
Even for Python 3? Then to be honest, I do not know what to change. On my system (and probably at NERSC), the `--embed` option is necessary as otherwise `-lpython.x.y` link option is **not** included and the whole build fails because the subsequent attempt to extract the Python library name fails. I am wondering if they added it for a later Python 3.x version. What version is installed on LC?

Reply:
Some googling, it seems to have changed with Python 3.8: https://bugs.python.org/issue36721, so I guess we should add the option based on the exact Python version and not just Python 2 vs. Python 3.